packages/jaeger-ui/src/components/SearchTracePage/SearchForm.jsx

```diff
@@ -257,338 +257,338 @@ export function submitForm(fields, searchTraces) {
   });
 }
 
-export class SearchFormImpl extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      formData: {
-        service: this.props.initialValues?.service,
-        operation: this.props.initialValues?.operation,
-        tags: this.props.initialValues?.tags,
-        lookback: this.props.initialValues?.lookback,
-        startDate: this.props.initialValues?.startDate,
-        startDateTime: this.props.initialValues?.startDateTime,
-        endDate: this.props.initialValues?.endDate,
-        endDateTime: this.props.initialValues?.endDateTime,
-        minDuration: this.props.initialValues?.minDuration,
-        maxDuration: this.props.initialValues?.maxDuration,
-        resultsLimit: this.props.initialValues?.resultsLimit,
-      },
-    };
-  }
+export function SearchFormImpl(props) {
+  const {
+    invalid,
+    searchMaxLookback,
+    services,
+    submitting,
+    initialValues,
+    changeServiceHandler,
+    submitFormHandler,
+    handleChange: propHandleChange,
+  } = props;
+
+  const [formData, setFormData] = React.useState({
+    service: initialValues?.service,
```

Comment:
what is the value of expanding `initialValues` like this instead of just storing as an object? It's not like `setFormData` provides a way to set individual fields in this struct, it's always all or nothing.